tests/_data/test_mask_semantic_segmentation_dataset.py

```diff
@@ -32,6 +33,12 @@
 from .. import helpers
 
 
+@pytest.fixture
+def mask_file_pattern() -> str:
+    """Standard mask file pattern for semantic segmentation tests."""
+    return "{image_path.parent.parent}/masks/{image_path.stem}.png"
```

Comment:
Don't create fixtures for constants. I would copy paste the constant to the relevant places, it makes the tests easier to read. Alternatively create a constant variable 
`MASK_FILE_PATTTERN="{image_path.parent.parent}/masks/{image_path.stem}.png"` but for tests it is nearly always better to hardcode these things.